lib/developer_portal/app/controllers/developer_portal/admin/account/payment_details_base_controller.rb

```diff
@@ -27,7 +27,7 @@ def edit
 
   def update
     current_account.updating_payment_detail = true
-    if current_account.update account_params
+    if current_account.update(account_params) && update_address_on_payment_gateway
```

Comment:
With this code, it could happen that the DB has been updated locally but the controller returns an error, which is confusing. We could enforce synchronicity by calling `current_account.update(account_params)` and `update_address_on_payment_gateway` inside a transaction. Or even like this:

```suggestion
    if update_address_on_payment_gateway && current_account.update(account_params)
```

Since I find `update_address_on_payment_gateway` much more likely to fail than Rails update.

Anyway, in both cases the user wouldn't be able to update their local database if something happens in Stripe side.